packages/benchmarks/src/textToDriver/nodeJsDriverBenchmark.ts

```diff
@@ -0,0 +1,167 @@
+import { runTextToDriverEval } from "./runTextToDriverEval";
+import { radiantModels } from "../radiantModels";
+import { assertEnvVars, MongoClient } from "mongodb-rag-core";
+import { NODE_JS_PROMPTS } from "./generateDriverCode/languagePrompts/nodeJs";
+import OpenAI, { AzureOpenAI } from "openai";
+import { TEXT_TO_DRIVER_ENV_VARS } from "./TextToDriverEnvVars";
+import { RADIANT_ENV_VARS } from "../envVars";
+import { wrapOpenAI } from "braintrust";
+import PromisePool from "@supercharge/promise-pool";
+async function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+async function main() {
+  const {
+    BRAINTRUST_API_KEY,
+    BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME,
+    MONGODB_TEXT_TO_DRIVER_CONNECTION_URI,
+    RADIANT_API_KEY,
+    RADIANT_ENDPOINT,
+    MONGODB_AUTH_COOKIE,
+  } = assertEnvVars({ ...TEXT_TO_DRIVER_ENV_VARS, ...RADIANT_ENV_VARS });
+  const projectName = BRAINTRUST_TEXT_TO_DRIVER_PROJECT_NAME;
+  const datasetName = "text-to-query-results";
+  const maxConcurrency = 3;
+  const timeout = 60000;
+
+  const prompts = NODE_JS_PROMPTS.systemPrompts;
+  const mongoClient = new MongoClient(MONGODB_TEXT_TO_DRIVER_CONNECTION_URI);
+
+  const SAMPLE_DOCUMENT_LIMIT = 2;
+
+  try {
+    await mongoClient.connect();
+    await sleep(500);
+    const modelExperiments = radiantModels.map((modelInfo) => {
+      const modelExperiments = [];
+      for (const promptType of Object.keys(prompts)) {
+        for (const generateCollectionSchemas of [true, false]) {
+          modelExperiments.push({
+            modelInfo,
+            promptType,
+            generateCollectionSchemas,
+          });
+        }
+      }
+      return modelExperiments;
+    });
+    // Process models in parallel
+    await PromisePool.for(modelExperiments)
+      .withConcurrency(4)
+      .process(async (modelInfos) => {
+        await PromisePool.for(modelInfos)
+          .withConcurrency(1)
```

Comment:
- Where are these concurrency values coming from?
- Could we flatten this to just use one promise pool instead of two?
  - A pool with concurrency one will likely perform worse than a regular `for...of` loop (though also likely negligible relative to LLM latency)

Reply:
no, we want to run all the models in parallel while running 1 experiment at a time per model.